### PR TITLE
Migrate from JUnit 4 to JUnit 5

### LINKED ISSUE #1331: Migrate from JUnit 4 to JUnit 5 for the kubernetes-server-mocks
More and more systems are migrating to junit5, it would be nice to have the server-mocks provide native JUnit5 suppoert

### PR DESCRIPTION
**Pull request by kunal-kushwaha**
Fixes #1331

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

**Comment by rohanKanojia** [comment]
@kunal-kushwaha : Could you please resolve conflicts here also?

File: kubernetes-client/pom.xml [hunk 1]
```diff
@@ -267,14 +267,36 @@
...
+    <dependency>
+      <groupId>org.junit.platform</groupId>
+      <artifactId>junit-platform-runner</artifactId>
+      <version>${junit.platform.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-migrationsupport</artifactId>
       <version>${junit.version}</version>
```
Comment (rohanKanojia) [inline kubernetes-client/pom.xml]: Did you mean `junit.version` here or `junit.platform.version`

File: kubernetes-examples/pom.xml [hunk 2]
```diff
@@ -53,8 +53,32 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.junit.platform</groupId>
+      <artifactId>junit-platform-runner</artifactId>
+      <version>${junit.platform.version}</version>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-migrationsupport</artifactId>
```
Comment (rohanKanojia) [inline kubernetes-examples/pom.xml]: again, did you really mean to use `junit.version` here?

**Comment by rohanKanojia** [comment]
@ricardozanini : I see that you wanted to contribute to this issue. Would really appreciate if you could help to review this PR ;-) . This PR is sent by a Google summer of code student so would really appreciate if you contribute here(if you have time of course ;-) )

File: kubernetes-itests/pom.xml [hunk 3]
```diff
@@ -34,6 +34,7 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <junit.version>4.12</junit.version>
```
Comment (rohanKanojia) [inline kubernetes-itests/pom.xml]: why are you forcing junit 4 here? You can refactor it to use junit 5 packages also, right? Or did you face some problems with arquillian? Maybe arquillian is not compatible with Junit 5? Maybe we need to update arquilian also....

File: kubernetes-client/pom.xml (hunk 1 above)
Comment (kunal-kushwaha) [reply kubernetes-client/pom.xml]: @rohanKanojia: I changed <junit.version> to 5.5.1 used in vintage and jupiter.

File: kubernetes-examples/pom.xml (hunk 2 above)
Comment (kunal-kushwaha) [reply kubernetes-examples/pom.xml]: @rohanKanojia: Yes, it's the latest version 5.5.1 for org.junit.jupiter.

File: kubernetes-itests/pom.xml (hunk 3 above)
Comment (kunal-kushwaha) [reply kubernetes-itests/pom.xml]: @rohanKanojia: In integration tests for @RunWith(ArquillianConditionalRunner.class), I replaced RunWith with ExtendWith for JUnit5 but it was showing compatibility error.

File: pom.xml [hunk 4]
```diff
@@ -298,7 +325,7 @@
             <version>2.1.1.Final</version>
```
Comment (ricardozanini) [inline pom.xml]: properties?

File: extensions/knative/client/pom.xml [hunk 5]
```diff
@@ -66,11 +66,36 @@
...
-        </dependency>
+      <dependency>
+        <groupId>org.junit.jupiter</groupId>
+        <artifactId>junit-jupiter-engine</artifactId>
+        <version>${junit.version}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.junit.vintage</groupId>
+        <artifactId>junit-vintage-engine</artifactId>
```
[the remark this answers is not in the document]
Comment (kunal-kushwaha) [reply extensions/knative/client/pom.xml]: Hi, I put it just for testing but it's not required. Fixed.

File: kubernetes-model/kubernetes-model/src/test/java/io/fabric8/kubernetes/api/model/ValidationTest.java [hunk 6]
```diff
@@ -27,9 +29,14 @@ public void testNodeName() throws JsonProcessingException {
     new io.fabric8.kubernetes.api.model.NodeBuilder().withNewMetadata().withName("1.0.0.1").and().build();
   }
 
-  @Test(expected = ConstraintViolationException.class)
+  @Test
   public void testIllegalNodeName() throws JsonProcessingException {
-    new io.fabric8.kubernetes.api.model.NodeBuilder().withNewMetadata().withName("..").and().build();
+    Assertions.assertThrows(ConstraintViolationException.class, new Executable() {
```
[the remark this answers is not in the document]
Comment (kunal-kushwaha) [reply kubernetes-model/kubernetes-model/src/test/java/io/fabric8/kubernetes/api/model/ValidationTest.java]: Actually, the source version in the model was set to 1.7 which does not allow Lambda Expressions. Fixed.

File: kubernetes-itests/pom.xml (hunk 3 above)
Comment (kunal-kushwaha) [reply kubernetes-itests/pom.xml]: @rohanKanojia @dev-gaur: Can you look into this when you get the time and provide suggestions? 
The problem I was facing for itests was with @RunWith(ArquillianConditionalRunner.class)

Comment (kunal-kushwaha) [reply kubernetes-itests/pom.xml]: @rohanKanojia @dev-gaur @ricardozanini: Can you look into this when you get the time and provide suggestions? 
The problem I was facing for itests was with @RunWith(ArquillianConditionalRunner.class)

**Comment by rohanKanojia** [comment]
@kunal-kushwaha : Could you please rebase your branch with latest master? I think we are good to merge this.

File: CHANGELOG.md [hunk 7]
```diff
@@ -35,7 +35,8 @@
    * Removed deprecated KubernetesKind enum
 
   Dependency Upgrade
-
+    * Fix #1331: Migrated from JUnit 4 to JUnit 5
```
Comment (rohanKanojia) [inline CHANGELOG.md]: This should go in 4.4-SNAPSHOT section

**Comment by kunal-kushwaha** [comment]
@rohanKanojia: fixed conflicts

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**